### DISCUSSION TITLE
Use modern, multiplatform solution like Sentry for error and crash reporting?

### DISCUSSION THREAD

**Discussion by Pentaphon** [transmission/transmission#3417] [Ideas]
https://github.com/getsentry/sentry-native

Now that Transmission has been re-factored to C++ and will soon go to more users than ever before since it is out of "preview" status on Windows and the project will be getting more contributors than ever before...perhaps it's time to look into a modern, multiplatform crash logging solution like Sentry to ensure that logs are properly reported. 

Thoughts?

**Comment by ckerr** [comment]
Maybe? The way we do crash reporting now is pretty crude.

I have three possible issues with this:

- Transmission advertises itself as a no-tracking privacy-focused torrent client; we don't track website hits, we don't count users, we don't do anything. Using some crash reporter seems like it would go against that. We could make it opt-in instead of opt-out, I suppose?

- I personally don't enjoy sysadmin work and whenever something like this comes up that requires a service to be running somewhere I try to offload that task to some zero-effort provider (e.g. hosting issues and source code at GitHub) or another person (note how much better @mikedld is at fixing CI issues).

- Someone would need to volunteer to add this to the macOS and Windows builds.

If all these could be worked out, I'd tentatively be OK with trying it out.

**Reply by Pentaphon** [reply]
> Maybe? The way we do crash reporting now is pretty crude.

I'm just tossing out ideas considering that the scope of the project has gone beyond just Linux and MacOS and we're playing in the much bigger sandbox of Github than the old Transmission issue tracker, with more contributions and bug reports than ever before. 

If there's anything "more in the middle" for a solution, people are welcome to suggest it. I suggested Sentry since a lot of mobile and desktop applications are using it and it's open source and is free for open source projects.

https://sentry.io/for/open-source/

I just think the project's weakest aspect right now is reporting crashes and errors so if there's anything that can be done to make this process more organized and accurate, the project can avoid being the huge 2,800 issue mess that qBittorrent is currently experiencing where people just pile on issue after issue in such a vague manner that they never seem to get resolved.
